en/docs/learn/api-controller/advanced-topics/configuring-different-endpoint-types.md

```diff
@@ -0,0 +1,255 @@
+
+API Manager supports four (4) main types of endpoints as follows.
+
+1. HTTP/REST Endpoints
+2. HTTP/SOAP Endpoints
+3. Dynamic Endpoints
+4. AWS Lambda Endpoints
+
+!!! info
+    The following fields should be specified as your requirement when setting up different endpoints in `api_params.yaml` file.
+    
+    -   **endpointType**
+        
+        To specify the type of the endpoint. Values can be `rest`, `soap`, `dynamic` or `aws`. (For HTTP/REST Endpoints, you can specify the type as `rest`. But even if you did not specify the field `endpointType` at all in the `api_params.yaml` file, by default it will take the type as `rest`)
+    
+    -   **endpointRoutingPolicy**
+    
+        To specify the routing policy to be used. Values can be `load_balanced` or `failover`.
+    
+    -   **loadBalanceEndpoints** field should be specified if the `endpointRoutingPolicy` is `load_balanced`. This field contains the following.
+
+        - `production`: An array which consists the multiple production endpoints
+        - `sandbox`: An array which consists the multiple sandbox endpoints
+        - `sessionManagement` (Optional): Values can be `none`, `transport`, `soap`, `simpleClientSession` and if not specified the default value would be `transport`
+        - `sessionTimeout` (Optional): The number of milliseconds after which the session would time out
+
+    -   **failoverEndpoints** field should be specified if the `endpointRoutingPolicy` is `failover`. This field contains the following.
+
+        - `production`: The primary production endpoint (not an array)
+        - `sandbox`: The primary sandbox endpoint (not an array)
+        - `productionFailovers`: An array which consists failover production endpoints
+        - `sandboxFailovers`: An array which consists failover sandbox endpoints
+
+    -   **awsLambdaEndpoints** field should be specified if the `endpointType` is `aws`. This field contains the following.
+
+        - `accessMethod`: The access method of awslambda endpoint. Mandatory to specify. (Values can be `role_supplied` or `stored`)
+        - `amznRegion`: Region where the endpoint is located. Can be chosen from [here](https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints)
+        - `amznAccessKey`: Access Key for endpoint
+        - `amznSecretKey`: Access Secret for endpoint
+
+Let us discuss configuring each of the above endpoint type with examples, seperately, in the following sections.
+
+### HTTP/REST Endpoints
+
+This can be subdivided into three (3) main scenarios.
+
+#### 1. HTTP/REST Endpoint - Without load balancing or failover
+
+The following is an example `api_params.yaml` file for this scenario. (Note that you can ignore specifying the `endpointType` field if you want, because by default it will contain the value as `rest`.)
```

Comment:
```suggestion
The following is an example `api_params.yaml` file for this scenario. (Note that you can ignore specifying the `endpointType` field if you want, because by default it contains the value as `rest`.)
```